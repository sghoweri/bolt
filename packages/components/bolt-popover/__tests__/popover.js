import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
} from '../../../../packages-extras/testing/testing-helpers';
import schema from '../popover.schema';
const { placement, spacing, theme, uuid } = schema.properties;
const timeout = 120000;

describe('<bolt-popover> Component', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  // Basic Usage
  test('basic usage', async () => {
    const results = await render('@bolt-components-popover/popover.twig', {
      trigger: '<bolt-button>This triggers a popover</bolt-button>',
      content:
        'This is the content of the popover with a <bolt-link url="https://pega.com">call to action</bolt-link>.',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Props
  placement.enum.forEach(async placementChoice => {
    test(`content placement: ${placementChoice}`, async () => {
      const results = await render('@bolt-components-popover/popover.twig', {
        trigger: '<bolt-button>This triggers a popover</bolt-button>',
        content:
          'This is the content of the popover with a <bolt-link url="https://pega.com">call to action</bolt-link>.',
        placement: placementChoice,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  spacing.enum.forEach(async spacingChoice => {
    test(`content spacing: ${spacingChoice}`, async () => {
      const results = await render('@bolt-components-popover/popover.twig', {
        trigger: '<bolt-button>This triggers a popover</bolt-button>',
        content:
          'This is the content of the popover with a <bolt-link url="https://pega.com">call to action</bolt-link>.',
        spacing: spacingChoice,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  theme.enum.forEach(async themeChoice => {
    test(`content theme: ${themeChoice}`, async () => {
      const results = await render('@bolt-components-popover/popover.twig', {
        trigger: '<bolt-button>This triggers a popover</bolt-button>',
        content:
          'This is the content of the popover with a <bolt-link url="https://pega.com">call to action</bolt-link>.',
        theme: themeChoice,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  test(`UUID of the popover`, async () => {
    const results = await render('@bolt-components-popover/popover.twig', {
      trigger: '<bolt-button>This triggers a popover</bolt-button>',
      content:
        'This is the content of the popover with a <bolt-link url="https://pega.com">call to action</bolt-link>.',
      uuid: 'custom-unique-id',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
