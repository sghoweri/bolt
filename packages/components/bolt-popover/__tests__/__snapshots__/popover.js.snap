// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-popover> Component UUID of the popover 1`] = `
<bolt-popover uuid="custom-unique-id">
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-custom-unique-id">
      <a href="#js-bolt-popover-custom-unique-id"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-custom-unique-id"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-custom-unique-id"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component basic usage 1`] = `
<bolt-popover uuid="12345">
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: auto 1`] = `
<bolt-popover placement="auto"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: bottom 1`] = `
<bolt-popover placement="bottom"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: bottom-end 1`] = `
<bolt-popover placement="bottom-end"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: bottom-start 1`] = `
<bolt-popover placement="bottom-start"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: left 1`] = `
<bolt-popover placement="left"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: left-end 1`] = `
<bolt-popover placement="left-end"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: left-start 1`] = `
<bolt-popover placement="left-start"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: right 1`] = `
<bolt-popover placement="right"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: right-end 1`] = `
<bolt-popover placement="right-end"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: right-start 1`] = `
<bolt-popover placement="right-start"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: top 1`] = `
<bolt-popover placement="top"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: top-end 1`] = `
<bolt-popover placement="top-end"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content placement: top-start 1`] = `
<bolt-popover placement="top-start"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content spacing: medium 1`] = `
<bolt-popover spacing="medium"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-medium">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content spacing: none 1`] = `
<bolt-popover spacing="none"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap ">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content spacing: small 1`] = `
<bolt-popover spacing="small"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content spacing: xsmall 1`] = `
<bolt-popover spacing="xsmall"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-xsmall">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content theme: dark 1`] = `
<bolt-popover theme="dark"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble t-bolt-dark">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content theme: light 1`] = `
<bolt-popover theme="light"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble t-bolt-light">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content theme: none 1`] = `
<bolt-popover theme="none"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble ">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content theme: xdark 1`] = `
<bolt-popover theme="xdark"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble t-bolt-xdark">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;

exports[`<bolt-popover> Component content theme: xlight 1`] = `
<bolt-popover theme="xlight"
              uuid="12345"
>
  <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap c-bolt-popover--spacing-small">
    <span id="js-bolt-popover-trigger-12345">
      <a href="#js-bolt-popover-12345"
         class="c-bolt-popover__nojs-trigger"
         tabindex="-1"
      >
        <ssr-keep for="bolt-popover">
          <bolt-button>
            This triggers a popover
          </bolt-button>
        </ssr-keep>
      </a>
    </span>
    <span class="c-bolt-popover__no-js-content"
          id="js-bolt-popover-12345"
    >
      <div class="c-bolt-popover__content">
        <span class="c-bolt-popover__bubble t-bolt-xlight">
          <a href="#js-bolt-popover-trigger-12345"
             class="c-bolt-popover__nojs-close"
          >
            <span aria-hidden="true">
              &times;
            </span>
            <span class="u-bolt-visuallyhidden">
              Close popover
            </span>
          </a>
          <ssr-keep for="bolt-popover">
            <span slot="content">
              This is the content of the popover with a
              <bolt-link url="https://pega.com">
                call to action
              </bolt-link>
              .
            </span>
          </ssr-keep>
        </span>
      </div>
    </span>
  </div>
</bolt-popover>
`;
